lua/complementree/utils.lua

```diff
@@ -9,4 +13,71 @@ function M.cword(complete_item)
   return (complete_item.word or complete_item.abbr)
 end
 
+M.make_relative_path = function(path, root)
+  local baselen = #root
+  if path:sub(0, baselen) == root then
+    path = path:sub(baselen + 2)
+  end
+  return path
+end
+
+local scandir_default_opts = {
+  ignore_hidden = true,
+  max_depth = 8,
+  match_patterns = {},
+}
+
+M.scan_dir = function(root_dir, opts, list, depth)
```

Comment:
I tend to like it the other way:
```suggestion
function M.scan_dir(root_dir, opts, list, depth)
```